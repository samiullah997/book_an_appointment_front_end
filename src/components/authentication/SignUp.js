/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import SignupDetailsApi from '../../redux/reducer/user';

const SignUp = () => {
  const [loading, setLoading] = useState(false);
  const [showMessage, setShowMessage] = useState('');
  const dispatch = useDispatch();
  const state = useSelector((state) => state.UserReducer);
  const { signedUp } = state;
  const [email, setEmail] = useState('');
  const [name, setName] = useState('');
  const [password, setPassowrd] = useState('');
  const [confirmPassword, setConfirmPassowrd] = useState('');
  const [userName, setUserName] = useState('');

  const registerUser = (e) => {
    e.preventDefault();
    setLoading(true);
    if (email === '' || name === '' || password === '') return;
    const newUser = {
      email,
      name,
      username: userName,
      password,
      password_confirmation: confirmPassword,
    };
    dispatch(SignupDetailsApi(newUser));
    setEmail('');
    setName('');
    setPassowrd('');
    setUserName('');
    setConfirmPassowrd('');
  };

  useEffect(() => {
    if (signedUp === 'up') {
      setLoading(false);
      setShowMessage('Signed Up Successfully');
    } else if (signedUp === 'down') {
      setLoading(false);
      setShowMessage('Email Already Exists');
      setTimeout(() => setShowMessage(''), 3000);
    }
  }, [state]);

  return (
    <div className="flex h-screen w-full items-center justify-center bg-gray-900 bg-cover bg-no-repeat bg-image">
      <div className="rounded-xl bg-white bg-opacity-50 px-16 py-10 shadow-lg backdrop-blur-md max-sm:px-8">
        <div className="text-white">
          <div className="mb-8 flex flex-col items-center">
            <img src="./favicon.png" width="150" alt="" />
            <h1 className="mb-2 text-2xl">Buy Bikes</h1>
            <span className="text-white">Enter Login Details</span>
            <span className="text-xl text-green-800">{showMessage}</span>
          </div>
          <form onSubmit={registerUser}>
            <div className="mb-4 text-lg">
              <input className="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-75 shadow-lg outline-none backdrop-blur-md" required onChange={(e) => setName(e.target.value)} value={name} type="text" name="name" placeholder="David John" />
            </div>

            <div className="mb-4 text-lg">
              <input className="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-75 shadow-lg outline-none backdrop-blur-md" required onChange={(e) => setUserName(e.target.value)} value={userName} type="text" name="address" placeholder="username" />
            </div>

            <div className="mb-4 text-lg">
              <input className="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-75 shadow-lg outline-none backdrop-blur-md" required onChange={(e) => setEmail(e.target.value)} value={email} type="email" name="email" placeholder="john@gmail.com" />
            </div>

            <div className="mb-4 text-lg">
              <input className="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-75 shadow-lg outline-none backdrop-blur-md" required onChange={(e) => setPassowrd(e.target.value)} value={password} type="Password" name="password" placeholder="password" />
            </div>

            <div className="mb-4 text-lg">
              <input className="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-75 shadow-lg outline-none backdrop-blur-md" required onChange={(e) => setConfirmPassowrd(e.target.value)} value={confirmPassword} type="Password" name="confirm_password" placeholder="confirm password" />
            </div>
            <div className="mt-8 flex justify-center text-lg text-black">
              {loading ? (<img src="https://i.stack.imgur.com/kOnzy.gif" className="w-10 h-10" alt="loading" />) : (<button type="submit" className="rounded-3xl bg-black bg-opacity-50 px-10 py-2 text-white shadow-xl backdrop-blur-md transition-colors duration-300 hover:bg-yellow-600">Register</button>) }
            </div>
          </form>
          <p className="mt-8 text-xs font-light text-center text-gray-700">
            {' '}
            Already Register?
            {' '}
            <a
              href="./"
              className="font-medium text-purple-600 hover:underline"
            >
              login
            </a>
          </p>
        </div>
      </div>
    </div>
  );
};

export default SignUp;
