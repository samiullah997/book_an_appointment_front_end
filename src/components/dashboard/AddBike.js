import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { addBike } from '../../redux/reducer/bikeReducer';

const AddBike = () => {
  const navigate = useNavigate();
  const [name, setName] = useState('');
  const [brand, setBrand] = useState('');
  const [price, setPrice] = useState('');

  const [message, setMessage] = useState('');
  const [bikeType, setBikeType] = useState('');

  const [description, setDescription] = useState('');
  const [picture, setPicture] = useState('');
  const dispatch = useDispatch();
  const userData = localStorage.getItem('bookBikeUser');
  const user = JSON.parse(userData);
  if (!user) {
    navigate('/');
  }

  const submitBikeData = (e) => {
    e.preventDefault();
    const userData = localStorage.getItem('bookBikeUser');
    const user = JSON.parse(userData);
    if (!name || !brand || !price || !bikeType || !picture) return;
    const formData = {
      bike_image: picture,
      name,
      brand,
      daily_rate: price,
      description,
      bike_type: bikeType,
      user_id: user.userId,

    };
    dispatch(addBike(formData));
    setPicture('');
    setBrand('');
    setName('');
    setPrice('');
    setBikeType('');
    setDescription('');
    setMessage('Bike added successfully');
  };
  return (
    <div className="w-full h-full">
      <div className="flex flex-col items-center min-h-screen pt-6 sm:justify-center sm:pt-0 bg-gray-50">
        <div>
          <a href="/">
            <h3 className="text-4xl font-bold text-purple-600">
              Add Bike Details
            </h3>
          </a>
          <div className="flex flex-row justify-center items-center text-2xl font-bold text-green-700 mt-4">{message}</div>
        </div>
        <div className="w-full px-6 py-4 mt-6 overflow-hidden bg-white shadow-md sm:max-w-md sm:rounded-lg">
          <form onSubmit={submitBikeData}>
            <div>
              <div className="block text-sm font-medium text-gray-700 undefined">
                Picture
              </div>
              <div className="flex flex-col items-start">
                <input
                  type="text"
                  name="picture"
                  onChange={(e) => setPicture(e.target.value)}
                  value={picture}
                  className="px-2 block w-full h-10 mt-1 border border-gray-600 rounded-md shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50"
                  placeholder="url"
                  required
                />
              </div>
            </div>
            <div>
              <div className="block text-sm font-medium text-gray-700 undefined">
                Name
              </div>
              <div className="flex flex-col items-start">
                <input
                  type="text"
                  name="name"
                  onChange={(e) => setName(e.target.value)}
                  value={name}
                  className="px-2 block w-full h-10 mt-1 border border-gray-600 rounded-md shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50"
                  placeholder="Honda125"
                  required
                />
              </div>
            </div>
            <div className="mt-4">

              <div className="block text-sm font-medium text-gray-700 undefined">Brand</div>

              <div className="flex flex-col items-start">
                <input
                  type="text"
                  name="model"
                  onChange={(e) => setBrand(e.target.value)}
                  value={brand}
                  className="px-2 block w-full h-10 mt-1 border border-gray-600 rounded-md shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50"
                  placeholder="Bike Brand"
                  required
                />
              </div>
            </div>
            <div className="mt-4">
              <div className="block text-sm font-medium text-gray-700 undefined">
                Bike Type
              </div>
              <div className="flex flex-col items-start">
                <input
                  type="text"
                  name="engin_capacity"
                  onChange={(e) => setBikeType(e.target.value)}
                  value={bikeType}
                  className="px-2 block w-full h-10 mt-1 border border-gray-600 rounded-md shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50"
                  placeholder="Bike Type"
                  required
                />
              </div>
            </div>
            <div className="mt-4">
              <div className="block text-sm font-medium text-gray-700 undefined">
                Price
              </div>
              <div className="flex flex-col items-start">
                <input
                  type="number"
                  name="price"
                  onChange={(e) => setPrice(e.target.value)}
                  value={price}
                  className="px-2 block w-full h-10 mt-1 border border-gray-600 rounded-md shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50"
                  placeholder="$200"
                  required
                />
              </div>
            </div>
            <div className="mt-4">

              <div className="block text-sm font-medium text-gray-700 undefined">Description</div>

              <div className="flex flex-col items-start">
                <textarea
                  type="text"
                  name="description"
                  onChange={(e) => setDescription(e.target.value)}
                  value={description}
                  className="px-2 block w-full h-28 mt-1 border border-gray-600 rounded-md shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50"
                  placeholder="description here...."
                  required
                />
              </div>
            </div>
            <div className="flex items-center justify-center mt-4">
              <button
                type="submit"
                className="inline-flex items-center px-4 py-2 ml-4 text-xs font-semibold tracking-widest text-white uppercase transition duration-150 ease-in-out bg-gray-900 border border-transparent rounded-md active:bg-gray-900 false"
              >
                Add Bike
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};
export default AddBike;
