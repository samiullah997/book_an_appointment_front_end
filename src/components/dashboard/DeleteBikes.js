import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { removeBike } from '../../redux/reducer/bikeReducer';

const DeleteBikes = () => {
  const navigate = useNavigate();
  const [message, setMessage] = useState('');
  const dispatch = useDispatch();
  const bikeData = useSelector((state) => state.bikeReducer);
  const { bikes } = bikeData;
  const userData = localStorage.getItem('bookBikeUser');
  const user = JSON.parse(userData);
  if (!user) {
    navigate('/');
  }
  const reversed = [...bikes].reverse();
  if (!reversed) {
    return (
      <h6 className="text-center">Loading ...</h6>
    );
  }
  const deleteItem = (id) => {
    dispatch(removeBike(id));
    setMessage('Bike Deleted Successfully');
  };
  return (
    <div className="contianer flex flex-col items-center w-full h-full">
      <div className="text-2xl font-bold">List of All Bikes</div>
      <div className="text-xl text-red-900">{message}</div>
      <div className="container flex flex-col px-3 items-center h-full border">
        {
          reversed.map((item) => (
            <div className="w-full text-gray-900 bg-white py-2 rounded-lg  dark:text-white" key={item.id}>
              <div className="relative border dark:bg-gray-700 dark:border-gray-700 inline-flex items-center justify-between w-full px-4 py-2 text-sm font-medium border-b border-gray-200 rounded-t-lg hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:hover:bg-gray-600 dark:hover:text-white dark:focus:ring-gray-500 dark:focus:text-white">
                {item.name}
                <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth="1.5" stroke="currentColor" className="w-6 h-6 text-red-900 cursor-pointer" onClick={() => deleteItem(item.id)}>
                  <path strokeLinecap="round" strokeLinejoin="round" d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0" />
                </svg>
              </div>
            </div>
          ))
        }
      </div>
    </div>
  );
};

export default DeleteBikes;
