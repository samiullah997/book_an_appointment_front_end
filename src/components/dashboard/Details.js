import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { Link, useNavigate } from 'react-router-dom';
import { singleBike } from '../../redux/reducer/bikeReducer';

const Details = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const userData = localStorage.getItem('bookBikeUser');
  const user = JSON.parse(userData);
  if (!user) {
    navigate('/');
  }

  const bikeData = useSelector((state) => state.bikeReducer);
  const { bike } = bikeData;
  return (
    <div className="container flex flex-row w-full m-28">
      <div className="container flex justify-center py-2  items-center">

        <img src={bike.bike_image} className="w-full h-full" alt="details" />

      </div>
      <div className="conatiner w-full h-auto flex flex-col items-end py-4  px-30">
        <h1 className="text-2xl font-bold py-4">{bike.name}</h1>
        <div className="flex flex-row justify-between w-60 border bg-gray-100 px-2 py-2">
          <span>Price</span>
          <span>
            $

            {bike.daily_rate}
          </span>
        </div>
        <h1 className="text-2xl font-bold py-2  px-4">{bike.brand}</h1>
        <p className="text-2xl font-bold py-2  px-4">{bike.bike_type}</p>
        <Link to="/user/Reservation" onClick={() => dispatch(singleBike(bike.id))}>
          <button type="button" className="bg-green-500 mt-40 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full">Reserve</button>

        </Link>
      </div>
    </div>
  );
};
export default Details;
