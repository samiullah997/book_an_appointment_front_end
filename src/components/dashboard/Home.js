import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useNavigate } from 'react-router-dom';
import { singleBike } from '../../redux/reducer/bikeReducer';

const Home = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const [currentPage, setCurrentPage] = useState(1);
  const [postPerPage] = useState(3);
  const bikeData = useSelector((state) => state.bikeReducer);
  const userData = localStorage.getItem('bookBikeUser');
  const user = JSON.parse(userData);
  if (!user) {
    navigate('/');
  }
  const { bikes } = bikeData;
  const reversed = [...bikes].reverse();
  if (reversed.length === 0) {
    return (
      <div className="flex w-full h-full justify-center items-center">
        <img src="https://www.superiorlawncareusa.com/wp-content/uploads/2020/05/loading-gif-png-5.gif" className="w-10 h-10" alt="loading" />
      </div>
    );
  }

  const indexOfLastPost = currentPage * postPerPage;
  const indexOfFirstPost = indexOfLastPost - postPerPage;
  const currentPosts = reversed.slice(indexOfFirstPost, indexOfLastPost);
  const nextPage = () => {
    if (currentPage >= reversed.length / postPerPage) return;
    setCurrentPage(currentPage + 1);
  };
  const prevPage = () => {
    if (currentPage <= 1) return;
    setCurrentPage(currentPage - 1);
  };
  return (
    <section className="container overflow-scroll flex flex-col border h-full sm:h-200 sm:w-full items-center">
      <h2 className="mt-8 text-4xl font-bold text-purple-600">Hello Welcome to your Dashboard</h2>
      <h3 className="mt-2">List of your favourite Bikes</h3>
      <div className="">.........................</div>
      <div className="container w-full flex flex-row justify-center items-center h-full">
        <button className="bg-gray-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded-l-full" type="button" onClick={prevPage}>
          Prev
        </button>
        <div className="container w-4/5 flex flex-row px-3 space-x-3 justify-center items-center h-auto">
          {
            currentPosts.map((item) => (
              <div className="h-full w-2/5 bg-gray-50 rounded-lg shadow-md dark:bg-gray-800 dark:border-gray-700" key={item.id}>
                <img className="rounded-t-lg w-96 h-96 lg:w-50 lg:h-50 animate-wiggle" src={item.bike_image} alt="" />
                <div className="p-5">
                  <h5 className="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white">{item.name}</h5>
                  <p className="mb-3 font-normal text-gray-700 dark:text-gray-400">
                    $
                    {item.daily_rate}
                  </p>
                  <div className="flex flex-row justify-between">
                    <Link to={`/user/Details?id=${item.id}`} onClick={() => dispatch(singleBike(item.id))}>
                      <div className="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800">
                        View Details
                        <svg aria-hidden="true" className="w-4 h-4 ml-2 -mr-1" fillRule="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fillRule="evenodd" d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z" clipRule="evenodd" /></svg>

                      </div>
                    </Link>
                    <div />
                  </div>
                </div>
              </div>
            ))
          }
        </div>
        <button
          className="bg-gray-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded-r-full"
          type="button"
          onClick={nextPage}
        >
          Next
        </button>
      </div>
    </section>
  );
};
export default Home;
