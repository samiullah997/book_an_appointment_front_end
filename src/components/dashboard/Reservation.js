import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';

import { addReservation } from '../../redux/reducer/reservationReducer';

const Reservation = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const bikeData = useSelector((state) => state.bikeReducer);
  const { bike } = bikeData;
  const reservationData = useSelector((state) => state.reservationReducer);
  const { reservations } = reservationData;
  const userData = localStorage.getItem('bookBikeUser');
  const user = JSON.parse(userData);
  if (!user) {
    navigate('/');
  }
  const submitData = (e) => {
    e.preventDefault();
    const date = document.getElementById('date').value;
    const data = {
      user_id: user.userId,
      bike_id: bike.id,
      date,
      reserve: true,
    };
    dispatch(addReservation(data));
    document.getElementById('date').value = '';
    navigate('/user/dashboard');
  };

  const checkReservation = (id) => {
    const check = reservations.find((reservation) => reservation.bike_id === id);
    if (check) {
      return true;
    }
    return false;
  };

  return (
    <div className="w-full h-full bg-image border border-black">
      <form className="w-full h-full" onSubmit={submitData}>
        <div className="flex flex-col w-full h-full space-x-3 px-3 py-3 bg-green-700 bg-opacity-70 justify-center items-center">
          <img src={bike.bike_image} alt="bike" className="w-3/4 sm:w-1/4  mt-4" />
          <h1 className="font-bold text-2xl py-5 text-white">{bike.name}</h1>
          <hr />
          <div className="flex flex-col w-full h-full justify-center items-center">
            <input type="date" required name="date" id="date" className="bg-green-500 mt-4 w-3/4 sm:w-1/4 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full" />
            {checkReservation(bike.id) ? (<img src="https://media.istockphoto.com/id/828928236/vector/red-rubber-stamp-icon-on-transparent-background.jpg?b=1&s=612x612&w=0&k=20&c=gP3E8Zi5YAaKcqdH7ZOsrDdNzHURvQ2gJNdZORjH9PA=" className="h-10 w-30" alt="reserved" />) : <button type="submit" className="bg-green-500 mt-4 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full">Reserve</button>}

          </div>
        </div>
      </form>
    </div>
  );
};
export default Reservation;
