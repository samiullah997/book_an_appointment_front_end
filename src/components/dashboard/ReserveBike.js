/* eslint-disable max-len */
import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { addReservation } from '../../redux/reducer/reservationReducer';

const ReserveBike = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const bikeData = useSelector((state) => state.bikeReducer);
  const reservationData = useSelector((state) => state.reservationReducer);
  const { reservations } = reservationData;

  const [bikeDetail, setBikeDetails] = useState('');
  const { bikes } = bikeData;
  const setId = (e) => {
    const id = e.target.value;
    const selectedBike = bikes.find((bike) => bike.id === +id);
    setBikeDetails(selectedBike);
  };

  const checkReservation = (id) => {
    const check = reservations.find((reservation) => reservation.bike_id === id);
    if (check) {
      return true;
    }
    return false;
  };

  const userData = localStorage.getItem('bookBikeUser');
  const user = JSON.parse(userData);
  if (!user) {
    navigate('/');
  }

  const submitData = (e) => {
    e.preventDefault();
    const date = document.getElementById('date').value;
    const data = {
      user_id: user.userId,
      bike_id: bikeDetail.id,
      date,
      reserve: true,
    };
    dispatch(addReservation(data));
    document.getElementById('date').value = '';
  };
  return (
    <div className="w-full h-full bg-image border border-black">
      <form className="w-full h-full" onSubmit={submitData}>
        <div className="flex flex-col w-full h-full space-x-3 px-3 py-3 bg-green-700 bg-opacity-70 justify-center items-center">
          {bikeDetail && (
            <div className="flex flex-col justify-center items-center gap-2">
              <img src={bikeDetail.bike_image} alt="bike" className="w-3/4 sm:w-1/4" />
              <h1 className="font-bold text-2xl py-5 text-white">{bikeDetail.name}</h1>
            </div>
          )}

          <div className="flex flex-col w-full h-full justify-center items-center gap-2">
            <select onChange={setId} defaultValue="DEFAULT" className="bg-green-500 placeholder-gray-100 w-3/4 sm:w-1/4  hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full">
              <option disabled value="DEFAULT">Select Bike</option>
              {bikes.map((bike) => (
                <option key={bike.id} value={bike.id}>{bike.name}</option>
              ))}
            </select>
            <input type="date" required name="date" id="date" className="bg-green-500 w-3/4 sm:w-1/4 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full" />
            {checkReservation(bikeDetail.id) ? (<img src="https://media.istockphoto.com/id/828928236/vector/red-rubber-stamp-icon-on-transparent-background.jpg?b=1&s=612x612&w=0&k=20&c=gP3E8Zi5YAaKcqdH7ZOsrDdNzHURvQ2gJNdZORjH9PA=" className="h-10 w-30" alt="reserved" />) : <button type="submit" className="bg-green-500 mt-4 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full">Reserve</button>}
          </div>
        </div>
      </form>
    </div>
  );
};
export default ReserveBike;
