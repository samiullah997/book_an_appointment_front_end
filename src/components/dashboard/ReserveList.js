import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { removeReservation } from '../../redux/reducer/reservationReducer';

const ReserveList = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const reservationData = useSelector((state) => state.reservationReducer);
  const bikeData = useSelector((state) => state.bikeReducer);
  const { bikes } = bikeData;
  const { reservations } = reservationData;
  const userData = localStorage.getItem('bookBikeUser');
  const user = JSON.parse(userData);
  if (!user) {
    navigate('/');
  }

  const reversed = [...reservations].reverse();
  if (!reversed) {
    return (
      <h6 className="text-center">Loading ...</h6>
    );
  }

  return (
    <div className="contianer flex flex-col items-center w-full h-full overflow-scroll">
      <div>List of All Bikes</div>
      <div className="container flex flex-wrap px-3 justify-center items-center w-full gap-2 h-full border">
        {
                    reversed.map(((item) => {
                      const data = bikes.filter((bike) => bike.id === item.bike_id);
                      const bike = data;
                      return (
                        <div className="h-auto bg-gray-50 rounded-lg shadow-md dark:bg-gray-800 dark:border-gray-700" key={item.bike_id}>
                          <img className="rounded-t-lg w-96 h-96 lg:w-50 lg:h-50" src={bike[0].bike_image} alt="" />
                          <div className="p-5">
                            <h5 className="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white">{bike[0].name}</h5>
                            <p className="mb-3 font-normal text-gray-700 dark:text-gray-400">
                              $
                              {bike[0].daily_rate}
                            </p>
                            <div className="flex flex-row justify-between">
                              <button type="button" onClick={() => dispatch(removeReservation(item.id))} className="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 cursor-pointer">
                                Return
                                <svg aria-hidden="true" className="w-4 h-4 ml-2 -mr-1 text-white" fillRule="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fillRule="evenodd" d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z" clipRule="evenodd" /></svg>
                              </button>
                              <div />
                            </div>
                          </div>
                        </div>
                      );
                    }))
                }

      </div>
    </div>
  );
};
export default ReserveList;
