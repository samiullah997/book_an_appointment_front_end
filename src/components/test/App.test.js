import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import SignUp from '../authentication/SignUp';
import AddBike from '../dashboard/AddBike';
import Reservation from '../dashboard/Reservation';
import ReserveList from '../dashboard/ReserveList';
import ReserveBike from '../dashboard/ReserveBike';
import SidebarLayout from '../sidebar/sideBar';
import Details from '../dashboard/Details';
import DeleteBikes from '../dashboard/DeleteBikes';
import Home from '../dashboard/Home';

const MockApp = () => (
  <BrowserRouter>
    <Routes>
      <Route path="/signup" element={<SignUp />} />
      <Route path="/user/Reservation" element={<Reservation />} />
      <Route path="/user/ReserveBike" element={<ReserveBike />} />
      <Route element={<SidebarLayout />}>
        <Route path="/user/dashboard" element={<Home />} />
        <Route path="/user/AddBike" element={<AddBike />} />
        <Route path="/user/DeleteBikes" element={<DeleteBikes />} />
        <Route path="/user/Details" element={<Details />} />
        <Route path="/user/MyReservations" element={<ReserveList />} />
      </Route>
    </Routes>
  </BrowserRouter>
);

it('should match snapshot of the App component', () => {
  const tree = renderer.create(<MockApp />).toJSON();
  expect(tree).toMatchSnapshot();
});
