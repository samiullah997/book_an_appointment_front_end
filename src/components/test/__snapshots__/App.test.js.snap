// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot of the App component 1`] = `null`;
